File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    // 연결 갯수 구하기
    let answer = 0;
    
    ch = Array.from({length:n+1}, () => 0);
    
    
    function DFS(v){
        ch[v] = 1
        
        for (let i=0;i<n;i++) {
            if (i===v) continue;
            if (ch[i] != 1 && computers[v][i] == 1) {
                // 방문한 적 없고, 연결 되어 있다면
                DFS(i)
            }
        }
    }
    
    for (let i=0;i<n;i++) {
        if (ch[i] != 1) {
            // 방문한 적이 없다면
            DFS(i)
            answer += 1
        }
        
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque 

def solution(n, computers):
    answer = 0
    visit = [0] * (n+1)
    
    def bfs(node):
        queue = deque([(node)])
        visit[node] = 1
        
        while (queue):
            current = queue.popleft()
    
            for i in range(n):      
                if visit[i] != 1 and computers[current][i] == 1 :
                    # 방문하지 않은 경우 방문함
                    queue.append((i))
                    visit[i] = 1

    
    for i in range(n):
        
        if visit[i] != 1:
            bfs(i)
            answer += 1
        
      
    return answer